2024-06-06 16:23:31.821 +03:00 [ERR] Failed executing DbCommand (38ms) [Parameters=[@p0='?' (DbType = Guid), @p1='?' (DbType = DateTimeOffset), @p2='?' (Size = 4000), @p3='?' (DbType = DateTimeOffset), @p4='?' (Size = 4000), @p5='?' (Size = 4000), @p6='?' (DbType = Guid), @p7='?' (DbType = DateTimeOffset), @p8='?' (Size = 4000), @p9='?' (DbType = Boolean), @p10='?' (DbType = DateTimeOffset), @p11='?' (Size = 4000), @p12='?' (DbType = DateTimeOffset), @p13='?' (DbType = Guid), @p14='?' (Size = 4000), @p15='?' (DbType = Guid), @p16='?' (DbType = Int32), @p17='?' (Size = 4000)], CommandType='"Text"', CommandTimeout='30']
SET NOCOUNT ON;
INSERT INTO [Notifications] ([Id], [Created], [CreatedBy], [LastModified], [LastModifiedBy], [Name])
VALUES (@p0, @p1, @p2, @p3, @p4, @p5);
INSERT INTO [UserNotifications] ([Id], [Created], [CreatedBy], [IsRead], [LastModified], [LastModifiedBy], [NotificationDate], [NotificationId], [NotificationString], [ObjectId], [ObjectType], [UserId])
VALUES (@p6, @p7, @p8, @p9, @p10, @p11, @p12, @p13, @p14, @p15, @p16, @p17);
2024-06-06 16:23:31.989 +03:00 [ERR] An exception occurred in the database while saving changes for context type 'Offers.CleanArchitecture.Infrastructure.Data.AppDbContext'.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Cannot insert the value NULL into column 'Name', table 'OffersCleanArchitectureDb.dbo.Notifications'; column does not allow nulls. INSERT fails.
The INSERT statement conflicted with the FOREIGN KEY constraint "FK_UserNotifications_Notifications_NotificationId". The conflict occurred in database "OffersCleanArchitectureDb", table "dbo.Notifications", column 'Id'.
The statement has been terminated.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at Microsoft.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at Microsoft.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at Microsoft.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at Microsoft.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at Microsoft.Data.SqlClient.SqlDataReader.get_MetaData()
   at Microsoft.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption, Boolean shouldCacheForAlwaysEncrypted)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean isAsync, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry, String method)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method)
   at Microsoft.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior)
   at Microsoft.Data.SqlClient.SqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReader(RelationalCommandParameterObject parameterObject)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.Execute(IRelationalConnection connection)
ClientConnectionId:1f3072fe-d98d-4db8-9de9-4b55f427950b
Error Number:515,State:2,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.Execute(IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.SqlServer.Update.Internal.SqlServerModificationCommandBatch.Execute(IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.Execute(IEnumerable`1 commandBatches, IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Storage.RelationalDatabase.SaveChanges(IList`1 entries)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(IList`1 entriesToSave)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(StateManager stateManager, Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.<>c.<SaveChanges>b__112_0(DbContext _, ValueTuple`2 t)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges(Boolean acceptAllChangesOnSuccess)
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Cannot insert the value NULL into column 'Name', table 'OffersCleanArchitectureDb.dbo.Notifications'; column does not allow nulls. INSERT fails.
The INSERT statement conflicted with the FOREIGN KEY constraint "FK_UserNotifications_Notifications_NotificationId". The conflict occurred in database "OffersCleanArchitectureDb", table "dbo.Notifications", column 'Id'.
The statement has been terminated.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at Microsoft.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at Microsoft.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at Microsoft.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at Microsoft.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at Microsoft.Data.SqlClient.SqlDataReader.get_MetaData()
   at Microsoft.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption, Boolean shouldCacheForAlwaysEncrypted)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean isAsync, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry, String method)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method)
   at Microsoft.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior)
   at Microsoft.Data.SqlClient.SqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReader(RelationalCommandParameterObject parameterObject)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.Execute(IRelationalConnection connection)
ClientConnectionId:1f3072fe-d98d-4db8-9de9-4b55f427950b
Error Number:515,State:2,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.Execute(IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.SqlServer.Update.Internal.SqlServerModificationCommandBatch.Execute(IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.Execute(IEnumerable`1 commandBatches, IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Storage.RelationalDatabase.SaveChanges(IList`1 entries)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(IList`1 entriesToSave)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(StateManager stateManager, Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.<>c.<SaveChanges>b__112_0(DbContext _, ValueTuple`2 t)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges(Boolean acceptAllChangesOnSuccess)
2024-06-06 16:23:32.083 +03:00 [ERR] An error occurred while seeding the database.
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Cannot insert the value NULL into column 'Name', table 'OffersCleanArchitectureDb.dbo.Notifications'; column does not allow nulls. INSERT fails.
The INSERT statement conflicted with the FOREIGN KEY constraint "FK_UserNotifications_Notifications_NotificationId". The conflict occurred in database "OffersCleanArchitectureDb", table "dbo.Notifications", column 'Id'.
The statement has been terminated.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at Microsoft.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at Microsoft.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at Microsoft.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at Microsoft.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at Microsoft.Data.SqlClient.SqlDataReader.get_MetaData()
   at Microsoft.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption, Boolean shouldCacheForAlwaysEncrypted)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean isAsync, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry, String method)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method)
   at Microsoft.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior)
   at Microsoft.Data.SqlClient.SqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReader(RelationalCommandParameterObject parameterObject)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.Execute(IRelationalConnection connection)
ClientConnectionId:1f3072fe-d98d-4db8-9de9-4b55f427950b
Error Number:515,State:2,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.Execute(IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.SqlServer.Update.Internal.SqlServerModificationCommandBatch.Execute(IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.Execute(IEnumerable`1 commandBatches, IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Storage.RelationalDatabase.SaveChanges(IList`1 entries)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(IList`1 entriesToSave)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(StateManager stateManager, Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.<>c.<SaveChanges>b__112_0(DbContext _, ValueTuple`2 t)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges()
   at Offers.CleanArchitecture.Infrastructure.Data.ApplicationDbContextInitialiser.TrySeedAsync() in C:\Users\HP\source\repos\ahmedtabba\Offers.CleanArchitecture\src\Infrastructure\Data\ApplicationDbContextInitialiser.cs:line 380
   at Offers.CleanArchitecture.Infrastructure.Data.ApplicationDbContextInitialiser.SeedAsync() in C:\Users\HP\source\repos\ahmedtabba\Offers.CleanArchitecture\src\Infrastructure\Data\ApplicationDbContextInitialiser.cs:line 83
2024-06-06 16:32:18.719 +03:00 [ERR] ThreadPool.RunInThread() returned false!
2024-06-06 18:24:46.828 +03:00 [ERR] Job DEFAULT.NotificationCleanerJob threw an unhandled Exception: 
System.InvalidCastException: Identified object is not a String.
   at Quartz.Util.StringKeyDirtyFlagMap.GetString(String key)
   at Offers.CleanArchitecture.Infrastructure.BackGroundServices.Jobs.NotificationCleanerJob.Execute(IJobExecutionContext context) in C:\Users\HP\source\repos\ahmedtabba\Offers.CleanArchitecture\src\Infrastructure\BackGroundServices\Jobs\NotificationCleanerJob.cs:line 23
   at Quartz.Core.JobRunShell.Run(CancellationToken cancellationToken)
2024-06-06 18:24:47.139 +03:00 [ERR] Job DEFAULT.NotificationCleanerJob threw an exception.
Quartz.SchedulerException: Job threw an unhandled exception.
 ---> System.InvalidCastException: Identified object is not a String.
   at Quartz.Util.StringKeyDirtyFlagMap.GetString(String key)
   at Offers.CleanArchitecture.Infrastructure.BackGroundServices.Jobs.NotificationCleanerJob.Execute(IJobExecutionContext context) in C:\Users\HP\source\repos\ahmedtabba\Offers.CleanArchitecture\src\Infrastructure\BackGroundServices\Jobs\NotificationCleanerJob.cs:line 23
   at Quartz.Core.JobRunShell.Run(CancellationToken cancellationToken)
   --- End of inner exception stack trace --- [See nested exception: System.InvalidCastException: Identified object is not a String.
   at Quartz.Util.StringKeyDirtyFlagMap.GetString(String key)
   at Offers.CleanArchitecture.Infrastructure.BackGroundServices.Jobs.NotificationCleanerJob.Execute(IJobExecutionContext context) in C:\Users\HP\source\repos\ahmedtabba\Offers.CleanArchitecture\src\Infrastructure\BackGroundServices\Jobs\NotificationCleanerJob.cs:line 23
   at Quartz.Core.JobRunShell.Run(CancellationToken cancellationToken)]
2024-06-06 18:26:03.763 +03:00 [ERR] Job DEFAULT.NotificationCleanerJob threw an unhandled Exception: 
System.InvalidCastException: Identified object is not a String.
   at Quartz.Util.StringKeyDirtyFlagMap.GetString(String key)
   at Offers.CleanArchitecture.Infrastructure.BackGroundServices.Jobs.NotificationCleanerJob.Execute(IJobExecutionContext context) in C:\Users\HP\source\repos\ahmedtabba\Offers.CleanArchitecture\src\Infrastructure\BackGroundServices\Jobs\NotificationCleanerJob.cs:line 23
   at Quartz.Core.JobRunShell.Run(CancellationToken cancellationToken)
2024-06-06 18:26:03.860 +03:00 [ERR] Job DEFAULT.NotificationCleanerJob threw an exception.
Quartz.SchedulerException: Job threw an unhandled exception.
 ---> System.InvalidCastException: Identified object is not a String.
   at Quartz.Util.StringKeyDirtyFlagMap.GetString(String key)
   at Offers.CleanArchitecture.Infrastructure.BackGroundServices.Jobs.NotificationCleanerJob.Execute(IJobExecutionContext context) in C:\Users\HP\source\repos\ahmedtabba\Offers.CleanArchitecture\src\Infrastructure\BackGroundServices\Jobs\NotificationCleanerJob.cs:line 23
   at Quartz.Core.JobRunShell.Run(CancellationToken cancellationToken)
   --- End of inner exception stack trace --- [See nested exception: System.InvalidCastException: Identified object is not a String.
   at Quartz.Util.StringKeyDirtyFlagMap.GetString(String key)
   at Offers.CleanArchitecture.Infrastructure.BackGroundServices.Jobs.NotificationCleanerJob.Execute(IJobExecutionContext context) in C:\Users\HP\source\repos\ahmedtabba\Offers.CleanArchitecture\src\Infrastructure\BackGroundServices\Jobs\NotificationCleanerJob.cs:line 23
   at Quartz.Core.JobRunShell.Run(CancellationToken cancellationToken)]
2024-06-06 18:27:14.168 +03:00 [ERR] Job DEFAULT.NotificationCleanerJob threw an unhandled Exception: 
System.InvalidCastException: Identified object is not a String.
   at Quartz.Util.StringKeyDirtyFlagMap.GetString(String key)
   at Offers.CleanArchitecture.Infrastructure.BackGroundServices.Jobs.NotificationCleanerJob.Execute(IJobExecutionContext context) in C:\Users\HP\source\repos\ahmedtabba\Offers.CleanArchitecture\src\Infrastructure\BackGroundServices\Jobs\NotificationCleanerJob.cs:line 23
   at Quartz.Core.JobRunShell.Run(CancellationToken cancellationToken)
2024-06-06 18:27:14.311 +03:00 [ERR] Job DEFAULT.NotificationCleanerJob threw an exception.
Quartz.SchedulerException: Job threw an unhandled exception.
 ---> System.InvalidCastException: Identified object is not a String.
   at Quartz.Util.StringKeyDirtyFlagMap.GetString(String key)
   at Offers.CleanArchitecture.Infrastructure.BackGroundServices.Jobs.NotificationCleanerJob.Execute(IJobExecutionContext context) in C:\Users\HP\source\repos\ahmedtabba\Offers.CleanArchitecture\src\Infrastructure\BackGroundServices\Jobs\NotificationCleanerJob.cs:line 23
   at Quartz.Core.JobRunShell.Run(CancellationToken cancellationToken)
   --- End of inner exception stack trace --- [See nested exception: System.InvalidCastException: Identified object is not a String.
   at Quartz.Util.StringKeyDirtyFlagMap.GetString(String key)
   at Offers.CleanArchitecture.Infrastructure.BackGroundServices.Jobs.NotificationCleanerJob.Execute(IJobExecutionContext context) in C:\Users\HP\source\repos\ahmedtabba\Offers.CleanArchitecture\src\Infrastructure\BackGroundServices\Jobs\NotificationCleanerJob.cs:line 23
   at Quartz.Core.JobRunShell.Run(CancellationToken cancellationToken)]
